From eb3c19309938119ce3d448aaaa1e9b9201c8a38a Mon Sep 17 00:00:00 2001
From: Loc Vu <loc.vu.zn@renesas.com>
Date: Tue, 14 Jul 2020 13:19:29 +0700
Subject: [PATCH 030/476] ARM: dts: r8a77470: Add SYSC power-domain for
 r8a77470 nodes

- Add specified power-domains for CPU0 and CPU1 of R8A77470.
- Change <&cpg_clocks> to  <&sysc R8A77470_PD_ALWAYS_ON>
for power-domains of modules GPIO, SCIF, IRQC, SDHI, CMT.

Signed-off-by: Loc Vu <loc.vu.zn@renesas.com>
---
 arch/arm/boot/dts/r8a77470.dtsi | 38 ++++++++++++++++++++------------------
 1 file changed, 20 insertions(+), 18 deletions(-)

diff --git a/arch/arm/boot/dts/r8a77470.dtsi b/arch/arm/boot/dts/r8a77470.dtsi
index a3cbc6a..f3d9c07 100644
--- a/arch/arm/boot/dts/r8a77470.dtsi
+++ b/arch/arm/boot/dts/r8a77470.dtsi
@@ -34,6 +34,7 @@
 			reg = <0>;
 			clock-frequency = <1000000000>;
 			clocks = <&z2_clk>;
+			power-domains = <&sysc R8A77470_PD_CA7_CPU0>;
 			next-level-cache = <&L2_CA7>;
 		};
 
@@ -43,6 +44,7 @@
 			reg = <1>;
 			clock-frequency = <1000000000>;
 			clocks = <&z2_clk>;
+			power-domains = <&sysc R8A77470_PD_CA7_CPU1>;
 			next-level-cache = <&L2_CA7>;
 		};
 
@@ -104,7 +106,7 @@
 			#interrupt-cells = <2>;
 			interrupt-controller;
 			clocks = <&mstp9_clks R8A77470_CLK_GPIO0>;
-			power-domains = <&cpg_clocks>;
+			power-domains = <&sysc R8A77470_PD_ALWAYS_ON>;
 		};
 
 		gpio1: gpio@e6051000 {
@@ -118,7 +120,7 @@
 			#interrupt-cells = <2>;
 			interrupt-controller;
 			clocks = <&mstp9_clks R8A77470_CLK_GPIO1>;
-			power-domains = <&cpg_clocks>;
+			power-domains = <&sysc R8A77470_PD_ALWAYS_ON>;
 		};
 
 		gpio2: gpio@e6052000 {
@@ -132,7 +134,7 @@
 			#interrupt-cells = <2>;
 			interrupt-controller;
 			clocks = <&mstp9_clks R8A77470_CLK_GPIO2>;
-			power-domains = <&cpg_clocks>;
+			power-domains = <&sysc R8A77470_PD_ALWAYS_ON>;
 		};
 
 		gpio3: gpio@e6053000 {
@@ -147,7 +149,7 @@
 			#interrupt-cells = <2>;
 			interrupt-controller;
 			clocks = <&mstp9_clks R8A77470_CLK_GPIO3>;
-			power-domains = <&cpg_clocks>;
+			power-domains = <&sysc R8A77470_PD_ALWAYS_ON>;
 		};
 
 		gpio4: gpio@e6054000 {
@@ -161,7 +163,7 @@
 			#interrupt-cells = <2>;
 			interrupt-controller;
 			clocks = <&mstp9_clks R8A77470_CLK_GPIO4>;
-			power-domains = <&cpg_clocks>;
+			power-domains = <&sysc R8A77470_PD_ALWAYS_ON>;
 		};
 
 		gpio5: gpio@e6055000 {
@@ -175,7 +177,7 @@
 			#interrupt-cells = <2>;
 			interrupt-controller;
 			clocks = <&mstp9_clks R8A77470_CLK_GPIO5>;
-			power-domains = <&cpg_clocks>;
+			power-domains = <&sysc R8A77470_PD_ALWAYS_ON>;
 		};
 
 		pfc: pin-controller@e6060000 {
@@ -210,7 +212,7 @@
 				     <GIC_SPI 16 IRQ_TYPE_LEVEL_HIGH>,
 				     <GIC_SPI 17 IRQ_TYPE_LEVEL_HIGH>;
 			clocks = <&mstp4_clks R8A77470_CLK_IRQC>;
-			power-domains = <&cpg_clocks>;
+			power-domains = <&sysc R8A77470_PD_ALWAYS_ON>;
 		};
 
 		icram0:	sram@e63a0000 {
@@ -531,7 +533,7 @@
 			dmas = <&dmac0 0x29>, <&dmac0 0x2a>,
 			       <&dmac1 0x29>, <&dmac1 0x2a>;
 			dma-names = "tx", "rx", "tx", "rx";
-			power-domains = <&cpg_clocks>;
+			power-domains = <&sysc R8A77470_PD_ALWAYS_ON>;
 			status = "disabled";
 		};
 
@@ -545,7 +547,7 @@
 			dmas = <&dmac0 0x2d>, <&dmac0 0x2e>,
 			       <&dmac1 0x2d>, <&dmac1 0x2e>;
 			dma-names = "tx", "rx", "tx", "rx";
-			power-domains = <&cpg_clocks>;
+			power-domains = <&sysc R8A77470_PD_ALWAYS_ON>;
 			status = "disabled";
 		};
 
@@ -559,7 +561,7 @@
 			dmas = <&dmac0 0x2b>, <&dmac0 0x2c>,
 			       <&dmac1 0x2b>, <&dmac1 0x2c>;
 			dma-names = "tx", "rx", "tx", "rx";
-			power-domains = <&cpg_clocks>;
+			power-domains = <&sysc R8A77470_PD_ALWAYS_ON>;
 			status = "disabled";
 		};
 
@@ -573,7 +575,7 @@
 			dmas = <&dmac0 0x2f>, <&dmac0 0x30>,
 			       <&dmac1 0x2f>, <&dmac1 0x30>;
 			dma-names = "tx", "rx", "tx", "rx";
-			power-domains = <&cpg_clocks>;
+			power-domains = <&sysc R8A77470_PD_ALWAYS_ON>;
 			status = "disabled";
 		};
 
@@ -587,7 +589,7 @@
 			dmas = <&dmac0 0xfb>, <&dmac0 0xfc>,
 			       <&dmac1 0xfb>, <&dmac1 0xfc>;
 			dma-names = "tx", "rx", "tx", "rx";
-			power-domains = <&cpg_clocks>;
+			power-domains = <&sysc R8A77470_PD_ALWAYS_ON>;
 			status = "disabled";
 		};
 
@@ -602,7 +604,7 @@
 			dmas = <&dmac0 0xfd>, <&dmac0 0xfe>,
 			       <&dmac1 0xfd>, <&dmac1 0xfe>;
 			dma-names = "tx", "rx", "tx", "rx";
-			power-domains = <&cpg_clocks>;
+			power-domains = <&sysc R8A77470_PD_ALWAYS_ON>;
 			status = "disabled";
 		};
 
@@ -682,7 +684,7 @@
 			       <&dmac1 0xcd>, <&dmac1 0xce>;
 			dma-names = "tx", "rx", "tx", "rx";
 			max-frequency = <156000000>;
-			power-domains = <&cpg_clocks>;
+			power-domains = <&sysc R8A77470_PD_ALWAYS_ON>;
 			status = "disabled";
 		};
 
@@ -692,7 +694,7 @@
 			interrupts = <GIC_SPI 166 IRQ_TYPE_LEVEL_HIGH>;
 			clocks = <&mstp3_clks R8A77470_CLK_SDHI1>;
 			max-frequency = <156000000>;
-			power-domains = <&cpg_clocks>;
+			power-domains = <&sysc R8A77470_PD_ALWAYS_ON>;
 			status = "disabled";
 		};
 
@@ -706,7 +708,7 @@
 			       <&dmac1 0xd3>, <&dmac1 0xd4>;
 			dma-names = "tx", "rx", "tx", "rx";
 			max-frequency = <78000000>;
-			power-domains = <&cpg_clocks>;
+			power-domains = <&sysc R8A77470_PD_ALWAYS_ON>;
 			status = "disabled";
 		};
 
@@ -731,7 +733,7 @@
 				     <GIC_SPI 143 IRQ_TYPE_LEVEL_HIGH>;
 			clocks = <&mstp1_clks R8A77470_CLK_CMT0>;
 			clock-names = "fck";
-			power-domains = <&cpg_clocks>;
+			power-domains = <&sysc R8A77470_PD_ALWAYS_ON>;
 
 			renesas,channels-mask = <0x60>;
 
@@ -752,7 +754,7 @@
 				     <GIC_SPI 127 IRQ_TYPE_LEVEL_HIGH>;
 			clocks = <&mstp3_clks R8A77470_CLK_CMT1>;
 			clock-names = "fck";
-			power-domains = <&cpg_clocks>;
+			power-domains = <&sysc R8A77470_PD_ALWAYS_ON>;
 
 			renesas,channels-mask = <0xff>;
 
-- 
2.7.4

